core/templates/pages/exploration-player-page/services/exploration-recommendations.service.ts

```diff
@@ -72,8 +72,15 @@ export class ExplorationRecommendationsService {
       ExplorationRecommendationsService.isInEditorPage && (
         this.contextService.getEditorTabContext() ===
         ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW));
-    ExplorationRecommendationsService.explorationId = (
-      this.contextService.getExplorationId());
+
+    // This check is to ensure getExplorationId method won't be called
+    // when ExplorationRecommendationsService is called on UpgradedServices
+    // and then throw an error.
+    if (this.contextService.getPageContext() !== 'other' &&
+        this.contextService.getEditorTabContext() !== null) {
+      ExplorationRecommendationsService.explorationId = (
+        this.contextService.getExplorationId());
+    }
```

Comment:
This is a fix of #9339, it was failing because `getExplorationId` was being called when listing the file in `UpgradedServices.ts`

Reply:
Hi @marianazangrossi, looks like this change is causing e2e tests to fail. Can you run the tests locally or try to reproduce the steps in the e2e test manually to find out why this causes failures? Thanks!

Reply:
I'm seeing calls being made to `/explorehandler/recommendations/null` in many tests.